src/main/osd/osd_elements.c

```diff
@@ -312,7 +312,7 @@ static void renderOsdEscRpmOrFreq(getEscRpmOrFreqFnPtr escFnPtr, osdElementParms
 }
 #endif
 
-#if defined(USE_ADC_INTERNAL) || defined(USE_ESC_SENSOR)
+#if defined(USE_ADC_INTERNAL) || defined(USE_ESC_SENSOR) || defined(USE_DSHOT_TELEMETRY)
```

Comment:
(maybe just remove the guard - linker is smart enough to remove the code if it is unused. And testing for any temperature source is a bit of overkill